zookeeper-api/src/main/java/org/apache/helix/zookeeper/impl/client/FederatedZkClient.java

```diff
@@ -19,345 +19,535 @@
  * under the License.
  */
 
+import java.util.Iterator;
 import java.util.List;
+import java.util.Map;
+import java.util.NoSuchElementException;
+import java.util.concurrent.ConcurrentHashMap;
 import java.util.concurrent.TimeUnit;
 
+import com.sun.istack.internal.NotNull;
```

Comment:
Can we not include this? I understand what you're trying to do here, but I don't think you're using this particular annotation the right way.

See https://stackoverflow.com/questions/42576760/where-is-notnull-in-java